### PR TITLE
Fix false positive in `use_build_context_synchronously`

### LINKED ISSUE #2878: False positive in `use_build_context_synchronously`
**Describe the issue**
There are false positives when writing awaits inside anonymous functions defined above statements that use the `BuildContext` while using the `use_build_context_synchronously` lint.

**To Reproduce**
The linter warns about the `await` inside the `onTap` function.
```dart
void build(BuildContext context) {
  final childWidget = _Button(
    onTap: () async {
      await Future<void>.delayed(Duration());
    },
  );
  return Container(
    color: Theme.of(context).accentColor,
    child: childWidget,
  );
}
```

**Expected behavior**
The `await` inside the onTap shouldn't be taken into account with this lint rule, because the body inside the function is not executed.

### PR DESCRIPTION
**Pull request by davidmartos96**
# Description

There are false positives when writing awaits inside anonymous functions defined above.

Fixes #2878

PS: First time contributing to the linter so I may be taking wrong assumptions here, but this works for the false positive cases I've encountered.

### CODE REVIEW

File: lib/src/rules/use_build_context_synchronously.dart [hunk 1]
```diff
@@ -85,6 +85,12 @@ class _AwaitVisitor extends RecursiveAstVisitor {
   void visitAwaitExpression(AwaitExpression node) {
     hasAwait = true;
   }
+
+  @override
+  visitBlockFunctionBody(BlockFunctionBody node) {
```
Comment (bwilkerson) [inline lib/src/rules/use_build_context_synchronously.dart:90]: You probably also want to visit `ExpressionFunctionBody` to stop at the `=>` form.

**Review by bwilkerson — COMMENTED** [review]
Welcome to the linter package!

File: lib/src/rules/use_build_context_synchronously.dart (hunk 1 above)
Comment (davidmartos96) [reply lib/src/rules/use_build_context_synchronously.dart:90]: You are right, thanks!

**Review by bwilkerson — APPROVED** [review]

**Comment by pq** [comment]
This is fantastic.  Thank you!

**Comment by pq** [comment]
@davidmartos96: assuming you're not a Googler, how can we attribute your contribution in `AUTHORS`?

**Review by pq — APPROVED** [review]
💎

**Comment by davidmartos96** [comment]
@pq Maybe something similar to this 😅:

Individuals:

David Martos <davidmartos96@gmail.com>

**Pull request merged**